R/batch.R

```diff
@@ -0,0 +1,33 @@
+##' Run a batch of reports.
+##' 
+##' Run one report multiple times with different sets of parameters.
+##'
+##' @param name Name of the report to run (see
+##'   \code{\link{orderly_list}}).  A leading \code{src/} will be
+##'   removed if provided, allowing easier use of autocomplete.
+##'   Alternatively, the default of \code{NULL} is useful if you have
+##'   already set the working directory to be the source directory.
+##'
+##' @param parameters Data frame of parameters passed to report. Each row
+##'   represents a parameter set to be passed to one report run.
+##'
+##' @param ... Additional args passed to \code{\link{orderly_run}}
+##'
+##' @seealso \code{\link{orderly_run}} for details of report running
+##'
+##' @export
+##' @return List of ids of newly created reports
+##' 
+##' @examples
+##' 
+##' path <- orderly::orderly_example("demo")
+##' params <- data.frame(nmin = c(0.2, 0.25))
+##' ids <- orderly::orderly_run("other", params, root = path)
+orderly_batch <- function(name = NULL, parameters = NULL, ...) {
+  ## Assert each column named? or a test at least if not to see what happens
+  batch_id <- ids::random_id()
+  ids <- lapply(seq_len(nrow(parameters)), function(row) {
```

Comment:
I am sure you have shown me a better way to loop over rows in a type safe way. I couldn't remember what it was or where - can you remember?

Reply:
there is no great way of doing this - I'd be inclined to make a utility function df2list:

```
df2list <- function(df) {
  unname(split(df, seq_len(nrow(df)))
}
```

and then vcapply over that, or you could vcapply over what you have already

Reply:
Nice - that looks tidy. Thanks